dist/s_version

```diff
@@ -18,42 +19,66 @@ while :
 	esac
 done
 
-# If the version hasn't changed and we're not forcing the issue, we're done.
-if test "$force" = no -a \
-    -f $m4dir/version.m4 -a \
-    -f $m4dir/version-set.m4 ; then
-	eval `grep '^VERSION_[A-Z]*=' $m4dir/version-set.m4`
-	if test x${WIREDTIGER_VERSION_MAJOR} = x${VERSION_MAJOR} -a \
-		x${WIREDTIGER_VERSION_MINOR} = x${VERSION_MINOR} -a \
-		x${WIREDTIGER_VERSION_PATCH} = x${VERSION_PATCH} ; then
-		exit 0
+GenAutoconfVersion(){
+	# If the version hasn't changed and we're not forcing the issue, we're done.
+	if test "$force" = no -a \
+		-f $m4dir/version.m4 -a \
+		-f $m4dir/version-set.m4 ; then
+		eval `grep '^VERSION_[A-Z]*=' $m4dir/version-set.m4`
+		if test x${WIREDTIGER_VERSION_MAJOR} = x${VERSION_MAJOR} -a \
+			x${WIREDTIGER_VERSION_MINOR} = x${VERSION_MINOR} -a \
+			x${WIREDTIGER_VERSION_PATCH} = x${VERSION_PATCH} ; then
+			return
+		fi
 	fi
-fi
-
-dotted_version=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}.${WIREDTIGER_VERSION_PATCH}
-echo "Building $m4dir/version.m4"
-cat <<EOF > $m4dir/version.m4
-dnl WiredTiger product version for AC_INIT.  Maintained by dist/s_version
-${dotted_version}
-EOF
-
-echo "Building $m4dir/version-set.m4"
-cat <<EOF > $m4dir/version-set.m4
-dnl build by dist/s_version
-
-VERSION_MAJOR=${WIREDTIGER_VERSION_MAJOR}
-VERSION_MINOR=${WIREDTIGER_VERSION_MINOR}
-VERSION_PATCH=${WIREDTIGER_VERSION_PATCH}
-VERSION_STRING='"${WIREDTIGER_VERSION_STRING}"'
-
-AC_SUBST(VERSION_MAJOR)
-AC_SUBST(VERSION_MINOR)
-AC_SUBST(VERSION_PATCH)
-AC_SUBST(VERSION_STRING)
-
-VERSION_NOPATCH=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}
-AC_SUBST(VERSION_NOPATCH)
-EOF
+	dotted_version=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}.${WIREDTIGER_VERSION_PATCH}
+	echo "Building $m4dir/version.m4"
+	cat > $m4dir/version.m4 <<-END
+	dnl WiredTiger product version for AC_INIT.  Maintained by dist/s_version
+	${dotted_version}
+	END
+
+	echo "Building $m4dir/version-set.m4"
+	cat > $m4dir/version-set.m4 <<-END
+	dnl build by dist/s_version
+
+	VERSION_MAJOR=${WIREDTIGER_VERSION_MAJOR}
+	VERSION_MINOR=${WIREDTIGER_VERSION_MINOR}
+	VERSION_PATCH=${WIREDTIGER_VERSION_PATCH}
+	VERSION_STRING='"${WIREDTIGER_VERSION_STRING}"'
+
+	AC_SUBST(VERSION_MAJOR)
+	AC_SUBST(VERSION_MINOR)
+	AC_SUBST(VERSION_PATCH)
+	AC_SUBST(VERSION_STRING)
+
+	VERSION_NOPATCH=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}
+	AC_SUBST(VERSION_NOPATCH)
+	END
+}
+
+GenCmakeVersion(){
+	if test "$force" = no -a \
+		-f $cmakedir/version.cmake ; then
+		eval `sed -n -e 's/^set(\(WT_VERSION_[A-Z]*\) \([0-9]*\))$/\1=\2/p' $cmakedir/version.cmake`
+		if test x${WIREDTIGER_VERSION_MAJOR} = x${WT_VERSION_MAJOR} -a \
+			x${WIREDTIGER_VERSION_MINOR} = x${WT_VERSION_MINOR} -a \
+			x${WIREDTIGER_VERSION_PATCH} = x${WT_VERSION_PATCH} ; then
+			exit 0
```

Comment:
Oh woops, this should be a `return` instead of `exit`

Reply:
Edited! :)